server/text_generation_server/utils/layers.py

```diff
@@ -447,12 +447,12 @@ def forward(self, x):
 
 
 class MedusaHead(torch.nn.Module):
-    def __init__(self, config, prefix, weights):
+    def __init__(self, config, medusa_config, prefix, weights):
         super().__init__()
         self.blocks = torch.nn.ModuleList(
             [
                 ResBlock(config, prefix=f"{prefix}.{i}", weights=weights)
-                for i in range(config["medusa_num_layers"])
+                for i in range(medusa_config["medusa_num_layers"])
```

Comment:
Which should probably take that information from `speculate` instead (to avoid loading layers which we're not going to use)

We can do in another PR